### PR TITLE
ZER-104 - Add ability to send real messages from Heroku

### PR DESCRIPTION
**Pull request by AnastasiiaBilynska**
dev
## JIRA

* [ZER-104](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=3177&projectKey=ZER&view=detail&selectedIssue=ZER-104)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'zerowastemailer@gmail.com'
```
Comment (olegsobchuk) [inline app/mailers/application_mailer.rb:4]: change `zerowastemailer@gmail.com'` to `no-reply@zerowaste.com`

File: config/environments/production.rb [hunk 2]
```diff
@@ -133,4 +133,20 @@
   # config.active_record
   #   .database_resolver_context = ActiveRecord::Middleware:
   #   :DatabaseSelector::Resolver::Session
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.default_url_options = {
+    host: 'heroku.com',
```
Comment (olegsobchuk) [inline config/environments/production.rb:140]: host => `zero-waste-project.herokuapp.com`

File: app/controllers/users/registrations_controller.rb [hunk 3]
```diff
@@ -3,9 +3,10 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
     def send_devise_notification(notification, *args)
+      UserMailer.with(user: @user).test_email.deliver_later
```
Comment (olegsobchuk) [inline app/controllers/users/registrations_controller.rb:6]: remove this line

File: app/mailers/user_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class UserTestUserMailer < ApplicationMailer
```
Comment (olegsobchuk) [inline app/mailers/user_mailer.rb:1]: you class name doesn't match file name

File: app/mailers/user_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    default from: 'no-reply@zerowaste.com'
+
+    def welcome_email
```
Comment (olegsobchuk) [inline app/mailers/user_mailer.rb:4]: remove this

Comment (olegsobchuk) [reply app/mailers/user_mailer.rb:4]: add `test_email` method

File: spec/mailers/previews/user_mailer_preview.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_test_user_mailer
+class UserTestUserMailerPreview < ActionMailer::Preview
```
Comment (olegsobchuk) [inline spec/mailers/previews/user_mailer_preview.rb:2]: remove this file

File: config/routes/sidekiq.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
+require "sidekiq/web"
```
Comment (olegsobchuk) [inline config/routes/sidekiq.rb:1]: remove this file

File: config/routes.rb [hunk 8]
```diff
@@ -2,11 +2,18 @@
...
 
+# Rails.application.routes.draw do
+#   authenticate :user do
+#     mount Sidekiq::Web => 'admins/sidekiq'
+#   end
+
 Rails.application.routes.draw do
-  authenticate :admin do
-    mount Sidekiq::Web => 'admins/sidekiq'
+  def draw(routes_name)
```
Comment (olegsobchuk) [inline config/routes.rb:11]: use [THIS](https://github.com/mperham/sidekiq/issues/4061#issuecomment-629830410) instead

File: spec/mailers/user_mailer_spec.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
```
Comment (olegsobchuk) [inline spec/mailers/user_mailer_spec.rb:1]: remove this if you don't need it

**Review by olegsobchuk — DISMISSED** [review]
LGTM

**Review by Pazykiv — DISMISSED** [review]

**Review by JuliaKovtun — APPROVED** [review]

**Review by Pazykiv — APPROVED** [review]

**Review by JuliaKovtun — APPROVED** [review]

**Pull request merged**